### ISSUE TITLE
Need a clearer example of a Many To Many relationship.

### ISSUE THREAD

**Issue by misakikuva** [requery/requery#198]
First of all thanks for this library, it seems to be amazing, but I am struggling with creating my entities.

The database I am working on looks roughly like [this](https://i.imgur.com/gd7oW5Q.png). Since I have a M:N relationship between Item and Purchase, I decided to create the intermediate table classes ItemPurchase, 

What I wanted to know is how to represent this relationship using Requery? Should I create an Entity for my ItemPurchase table and then create a OnetoMany relationship between it and the Item and Purchase Entities or I can use the ManyToMany relationship directly between my Item and Purchase Entities? How does the JunctionTable annotation works?

If you want I can create a gist for the entities that I have so far.

**Comment by npurushe** [comment]
Generally the junction table is created automatically between the two. I'm not sure why you need to store the totalprice in the junction table entry seems its more appropriate to be in the purchase, but if that is how it is suppose to be you can model it something like this:

```java
@Entity
abstract class Item {
...
@JunctionTable(type = ItemPurchase.class)
@ManyToMany
List<Purchase> purchases;
}
@Entity
abstract class Purchase {
...
@ManyToMany
List<Item> items;
}
@Entity
abstract class ItemPurchase {
    @ForeignKey(  references = Item.class  )
    @Key
    protected Integer itemId;

    @ForeignKey( references = Purchase.class  )
    @Key
    protected Integer purchaseId;

    protected Integer totalPrice;
...
}
```
You can omit the `type` from `JunctionTable` and the junction table class definition will be created for you automatically, but will not have any additional columns except what is required to join the the types.
You can also reference the test models https://github.com/requery/requery/tree/master/requery-test/src/main/java/io/requery/test which define various types of many to many relationships

**Comment by misakikuva** [comment]
Hey, thanks for the reply and sorry for not replying earlier. 

In the case of  doing what you suggested, adding the totalPrice field to the Purchase table, I should just create the Purchase and Item Entities and use the ManyToMany annotation in both, right? Also, does it matter where I put the JunctionTable annotation?

**Comment by npurushe** [comment]
Yes correct. You can put the JunctionTable on either side, it really only affects the auto generated table name between them.

**Comment by YuriHeupa** [comment]
And what about if I need to add a ordering field to a junction table (i.e. having a Playlist Queue saving Media order) with entities generated by interfaces? I can't manage how to define this, and how to insert those objects.

```java

@Entity
public interface Queue extends Persistable {
    @Key
    int getId();
    @ManyToMany
    @JunctionTable(type = QueueItem.class)
    Set<MediaEntity> getMediaList();
}

@Entity
public interface QueueItem extends Persistable {

    int getPosition();

    @ForeignKey(references = Media.class)
    @Key
    Integer getPodcastId();
    @ForeignKey(references = Queue.class)
    @Key
    Integer getQueueId();
}

@Entity
public interface Media extends Parcelable, Persistable {
    @Key
    int getId();
    String getUrl();
    @ManyToMany
    Set<QueueEntity> getQueues();
}
```

Any sugestions?

**Comment by npurushe** [comment]
Hey yes sorry that issue is fixed by #200 an update will be available soon
